delft/utilities/Transformer.py

```diff
@@ -0,0 +1,171 @@
+
+                    if "path-vocab" in transformer_configuration and os.path.isfile(
+                            transformer_configuration["path-vocab"]):
+                        self.local_vocab_file = transformer_configuration["path-vocab"]
+                    else:
+                        print("Missing vocab-file or not a file.")
+            else:
+                self.loading_method = LOADING_METHOD_HUGGINGFACE_NAME
+                print("No configuration for", self.name, "Loading from Hugging face.")
+        else:
+            self.loading_method = LOADING_METHOD_HUGGINGFACE_NAME
+            print("No configuration for", self.name, "Loading from Hugging face.")
+
+    def init_preprocessor(self, max_sequence_length: int,
+                       add_special_tokens: bool = True,
+                       add_prefix_space: bool = True,
+                       empty_features_vector: Iterable[int] =None,
+                       empty_char_vector: Iterable[int] = None) -> BERTPreprocessor:
+        """
+        Load the tokenizer according to the provided information, in case of missing configuration,
+        it will try to use huggingface as fallback solution.
+        """
+        if self.loading_method == LOADING_METHOD_HUGGINGFACE_NAME:
+            self.tokenizer = AutoTokenizer.from_pretrained(self.name,
+                                                           add_special_tokens=add_special_tokens,
+                                                           max_length=max_sequence_length,
+                                                           add_prefix_space=add_prefix_space)
+
+        elif self.loading_method == LOADING_METHOD_LOCAL_MODEL_DIR:
+            self.tokenizer = AutoTokenizer.from_pretrained(self.local_dir_path,
+                                                           add_special_tokens=add_special_tokens,
+                                                           max_length=max_sequence_length,
+                                                           add_prefix_space=add_prefix_space)
+        elif self.loading_method == LOADING_METHOD_PLAIN_MODEL:
+            self.tokenizer = BertTokenizer.from_pretrained(self.local_vocab_file)
+
+        elif self.loading_method == LOADING_METHOD_DELFT_MODEL:
+            config_path = os.path.join(".", self.local_dir_path, TRANSFORMER_CONFIG_FILE_NAME)
+            self.transformer_config = AutoConfig.from_pretrained(config_path)
+            self.tokenizer = AutoTokenizer.from_pretrained(os.path.join(self.local_dir_path, DEFAULT_TRANSFORMER_TOKENIZER_DIR), config=self.transformer_config)
+
+        self.bert_preprocessor = BERTPreprocessor(self.tokenizer, empty_features_vector, empty_char_vector)
+
+        return self.bert_preprocessor
+
+    def save_tokenizer(self, output_directory):
+        self.tokenizer.save_pretrained(output_directory)
+
+    def instantiate_layer(self, load_pretrained_weights=True) -> Union[object, TFAutoModel, TFBertModel]:
```

Comment:
I suppose the TFBertModel is not correct here? It could be other type of transformer models.  
This is a case where I would not introduce a typing here, because if the HuggingFace library changes the Object types, it will fail, while otherwise it would continue to work whatever Object they use, what do you think? 
Knowing explicitly this type is also not helpful as a user of the HuggingFace "magic" :)

Reply:
TFBertModel is returned in the case in line 164:

```
transformer_model = TFBertModel.from_config(self.transformer_config)
```

"if hugging face library changes the object type" => you mean if huggingface changes the returned objects in a new version?